### ISSUE TITLE
Time to bring up lightbulb seems to have regressed

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#26362]
I'm commonly seeing around 0.5-1.0 second for a lightblub to come up.  It's noticeable, without feeling utterly awful.  My guess is that there's some refactoring/analyzer running that's just taking a little too long, and htat's slowing down the whole experience.

This is **very** noticeable when typing a unbound type and hitting ctrl-dot to try to add a using.  I'm used to this being almost instant, and now it's just chugs.  

Note: i'm only on a *dual-core* laptop.  So it's quite possible people on the team aren't really feeling this if they're using beefy machines which can do lots of work in parallel.  But it's def gotten worse and i would love some attention here on determining if there are poorly behaving analyzers/refactorings that we can tweak to improve things.

**Comment by Neme12** [comment]
I agree. I've been seeing very long times and even dialog boxes with "Add using" recently. I don't remember if this used to be better, but I noticed these delays just today.

**Comment by CyrusNajmabadi** [comment]
Note: i think @Neme12 also has a fairly anemic laptop like me :)   Is that correct @Neme12 ?   Could some internal testing/perf traces be done on these types of machines to see what's up?

**Comment by Pilchie** [comment]
Telemetry doesn't *seem* to show a significant regression at the P95:

![image](https://user-images.githubusercontent.com/312835/39209369-858a6b16-47ba-11e8-9164-188c9b7b85d2.png)

**Comment by sharwell** [comment]
Can you link me to a feedback ticket created with these steps?
https://aka.ms/reportPerf

**Comment by CyrusNajmabadi** [comment]
Sure.  give me a few minutes.

**Comment by CyrusNajmabadi** [comment]
@sharwell have filed "extremely slow lightbulb #2".  I screwed up the original issue and didn't include the perf trace.  I tagged you in this item.  Note: i have no idea how to actually find this item or link you to it.  So i hope you can find it.

**Comment by sharwell** [comment]
https://developercommunity.visualstudio.com/content/problem/239525/extremely-slow-lightbulb-2.html

**Comment by CyrusNajmabadi** [comment]
Thanks!  Let me know if you need more data or want me to try anything else.

**Comment by sharwell** [comment]
Does the problem continue if you disable FSA for VB?

**Comment by CyrusNajmabadi** [comment]
Yes.  I turned off FSA already and still repro this after restart.  Sorry, i should have mentioned that.  Note: if you're seeing FSA on for VB... then something is very strange as my VS shows me: 

![image](https://user-images.githubusercontent.com/4564579/39211711-5c1e6bc2-47c1-11e8-8e18-05838df61918.png)

**Comment by CyrusNajmabadi** [comment]
@jcouv @dotnet/roslyn-compiler Did something regress here?  I'm just doing normal editing in a code body, but time to bring up lightbulbs has been seriously impacted (to the order of several seconds sometimes).  This is a pretty bad and very noticeable experience.

**Comment by CyrusNajmabadi** [comment]
@sharwell The perf is also bad in C# projects (though not as bad).  I'm filing another bug so you get a trace there.  Note: do you have a callstack there?  What codepath is leading to the WriteDocCommentXml path?

**Comment by CyrusNajmabadi** [comment]
Filed: ```Slow lightbulb perf (C# repro)```

**Comment by Neme12** [comment]
@sharwell I submitted a report too if you're interested to see if it's the same issue or not. It's not as slow as it was earlier though. Maybe I just didn't have the solution open long enough after all.

**Comment by CyrusNajmabadi** [comment]
@sharwell any data on the C# trace?

**Comment by CyrusNajmabadi** [comment]
@jcouv Can you help route the compiler issue accordingly.

Also, @sharwell Just to make sure, do those traces include other processes outside of VS?  I just want to make sure the cost of the OOP process is being measured as well and not just VS.  

Thanks!

**Comment by sharwell** [comment]
> any data on the C# trace?

It's appears to be similar, but much less severe

> Also, @sharwell Just to make sure, do those traces include other processes outside of VS? I just want to make sure the cost of the OOP process is being measured as well and not just VS.

They do

**Comment by CyrusNajmabadi** [comment]
@sharwell Note that i was still getting a +1-2 second pause for C#, so i consider that a pretty severe regression in performance.  Also, it would be useful to know what the root cause was since i'm guessing it wasn't the MislocatedDocumentationCommentFinder (which is VB only).  :)

**Comment by CyrusNajmabadi** [comment]
@jcouv Can this be assigned to someone appropriate on the compiler team?  This worries me that we might be about to ship some pretty bad behavior.  For context, lightbulbs are often taking 5-8 *seconds* on my machine to appear.  It's *terrible*.  

Note: i am only on a dual core laptop.  So it's possible something changes that exacerbated this and it's less noticeable for the beefy desktops people tend to have on the team.

**Comment by CyrusNajmabadi** [comment]
> It's appears to be similar, but much less severe

Note: it still feels pretty severe to me.  I measure lightbulbs for C# taking 4+ seconds.  I'm not sure what's happened, but it's really not good and feels like a def regression.  DO you have an analysis of what's taking the time in the c# trace?

**Comment by sharwell** [comment]
I haven't completed the analysis yet.

**Comment by jcouv** [comment]
Tagging @jaredpar and @OmarTawfik who did some investigation.

**Comment by CyrusNajmabadi** [comment]
This is still bad on the newest preview.  Can this get assigned to someone?  It worries me that this big a regression doesn't have any active eyes on things.

**Comment by OmarTawfik** [comment]
I've been looking at another issue regarding incremental parsing. I don't think these are related. However, I can take a look afterwards. @sharwell let me know if I can help.

**Comment by CyrusNajmabadi** [comment]
It's been two weeks... and this doesn't ahve someone assigned to it.  The perf hit on lightbulbs is really bad and very noticeable.  Can this be looked at, or at least assigned to someone so that it isn't lost to the ether?

**Comment by sharwell** [comment]
Duplicate of #39315
